src/player.rs

```diff
@@ -0,0 +1,47 @@
+//! This module contains a barebones loader for track files.
+use crate::track::Track;
+use std::collections::HashMap;
+use std::fs::File;
+use std::path::Path;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Failed to open file for reading track data")]
+    OpenTrackFile(#[source] std::io::Error),
+    #[error("Failed to deserialize track data")]
+    DeserializeTracks(#[source] bincode::Error),
+}
+
+/// A loader for track binary files dumped by [`Client::save_tracks`](crate::Client::save_tracks).
+///
+/// # Usage
+///
+/// After constructing, call [`Player::get_track`] to get tracks.
+/// Then call [`Track::get_value`] to get saved values at any given point in time.
+pub struct Player {
```

Comment:
This could then be called `RocketPlayer`.

Reply:
Yeah when naming types like crate::CrateType, after `use`ing, the type still tells everything about itself. Even though it's a bit redundant. I think your suggestion is the best option in this case